homeassistant/components/niko_home_control/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow for the Niko home control integration."""
+
+from __future__ import annotations
+
+import ipaddress
+from typing import Any
+
+from nikohomecontrol import NikoHomeControlConnection
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+
+from .const import DEFAULT_IP, DEFAULT_PORT, DOMAIN
+from .errors import CannotConnect, InvalidHost, InvalidPort
+
+
+async def validate_input(hass: HomeAssistant, data: dict) -> dict[str, str | int]:
+    """Validate the user input allows us to connect."""
+    host = data[CONF_HOST]
+    port = data[CONF_PORT]
+
+    try:
+        ipaddress.ip_address(host)
+    except ValueError:
+        raise InvalidHost from None
+
+    if port < 0 or port > 65535:
+        raise InvalidPort
```

Comment:
Oh I see. I think we have stuff for this in the config_validation for this, but afaik we don't actively look for this

Reply:
So I can replace it with something builtin is what your saying?

Reply:
On one hand, I think so yes, on the other hand, we hardly have these checks in general. So unless there is a big need we need them, we can omit them